### PR TITLE
Adding ARIA for calendar component

### LINKED ISSUE #6272: Calendar ARIA
## Is your feature request related to a problem? Please describe.
There's currently no ARIA attributes applied to the calendar (not in spec either). This makes screen readers announce partial information when navigating the calendar like "Clickable 27". 

## Describe the solution you'd like
The calendar should implement ARIA attributes per spec (see below) so each date can be announced when navigating with a screen reader.

## Describe alternatives you've considered
N/A

## Additional context
Per https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/datepicker-dialog.html the calendar should be using a grid pattern. Also some interesting abbreviation in the table headers.
Note:  This example might be outdated as it behaves a bit odd.

### PR DESCRIPTION
**Pull request by desig9stein**
Closes #6272

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/calendar/months-view/months-view.component.html [hunk 1]
```diff
@@ -1,8 +1,14 @@
 <div class="igx-calendar__body">
-    <div class="igx-calendar__body-row--wrap">
-        <div [igxCalendarMonth]="month" [date]="date" [attr.tabindex]="activeMonth === month.getMonth() ? 0 : -1" (onMonthSelection)="selectMonth($event)" [index]="i" *ngFor="let month of months; index as i; trackBy: monthTracker">
+    <div  class="igx-calendar__body-row--wrap">
+        <span
+            [attr.aria-label]="month | date: 'LLLL'"
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/calendar/months-view/months-view.component.html:4]: It seems `aria-label` must be combined with something that supports one. Like this, on its own applied to a span, at lest NVDA for me reads months as "Clickable Dec" ignoring the nice "December" label :(

**Review by zdrawku — DISMISSED** [review]
As we discussed with Damyan, his comment will be addressed with a future release

**Review by zdrawku — COMMENTED** [review]
@desig9stein could you please update the readme.md file and changelog if necessary

File: projects/igniteui-angular/src/lib/calendar/calendar.component.html [hunk 2]
```diff
@@ -4,49 +4,99 @@
...
-    <span tabindex="0" #monthsBtn (keydown)="onActiveViewYearKB(getViewDate(obj.index), $event, obj.index)" (click)="onActiveViewYear(getViewDate(obj.index), obj.index, $event)"
-        class="igx-calendar-picker__date">
-        {{ formattedMonth(getViewDate(obj.index)) }}
-    </span>
-    <span tabindex="0" #yearsBtn (keydown)="onActiveViewDecadeKB($event, getViewDate(obj.index), obj.index)" (click)="onActiveViewDecade(getViewDate(obj.index), obj.index)"
-        class="igx-calendar-picker__date">
-        {{ formattedYear(getViewDate(obj.index)) }}
-    </span>
+        <span class="igx-calendar__aria-off-screen" aria-live="polite">
+             {{ monthsViewNumber > 1 ? resourceStrings.first_picker + resourceStrings.of + ' ' + monthsViewNumber + ' ' + resourceStrings.starts_from + ' ' + (getViewDate(obj.index) | date: 'LLLL yyyy') : resourceStrings.igx_selected_month_is + (getViewDate(obj.index) | date: 'LLLL yyyy')}}
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/calendar/calendar.component.html:8]: Okay, this needs some adjusting - it's rendered for all month views (despite the text being "first of") and it's bound to month that changes for all of views when you navigate to previous/next. I realize Chromevox reads just one of those, but NVDA reads all in succession. For multiple months this only has to be rendered for the first one, right?

Also, there's no space between "first picker" and "of" and unless used elsewhere - you can make that a single locale string like `"First picker of {0} start from"`. There's no reason to translate just "of" out of context, might not even be possible in some languages.

File: projects/igniteui-angular/src/lib/calendar/calendar.component.html [hunk 3]
```diff
@@ -4,49 +4,99 @@
...
-            <igx-icon fontSet="material">keyboard_arrow_left</igx-icon>
+    <section class="igx-calendar-picker">
+        <span tabindex="0" class="igx-calendar__aria-off-screen">
+            <ng-container *ngIf="selection === 'multi'">
+                {{ resourceStrings.multi_selection }}
+            </ng-container>
+            <ng-container *ngIf="selection === 'range'">
+                {{ resourceStrings.range_selection }}
+            </ng-container>
+            calendar
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/calendar/calendar.component.html:58]: Was this supposed to be `resourceStrings.calendar`?

File: projects/igniteui-angular/src/lib/calendar/calendar.component.html [hunk 4]
```diff
@@ -4,49 +4,99 @@
...
+            {{ monthsViewNumber && monthsViewNumber > 1 ?  (resourceStrings.with + ' ' + monthsViewNumber + ' ' + resourceStrings.date_pickers) : ''}}
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/calendar/calendar.component.html:59]: Same thing as with `of`  just merge those into a single string like ` with {0} date pickers`

File: projects/igniteui-angular/src/lib/core/i18n/calendar-resources.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+export const CalendarResourceStringsEN: ICalendarResourceStrings = {
+ igx_calendar_previous_month: 'Previous Month',
+ igx_calendar_next_month: 'Next Month',
+ igx_calendar_select_month: 'Select Month',
+ igx_calendar_select_year: 'Select Year',
+ igx_calendar_range_start: 'Range start',
+ igx_calendar_range_end: 'Range end',
+ igx_selected_month_is: 'Selected month is',
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/core/i18n/calendar-resources.ts:26]: igx_selected_month? :) New component? It's either component prefixed (hint: they all should be) or not.

**Review by damyanpetev — CHANGES_REQUESTED** [review]
Requesting changes mainly for the localization strings comments, the rest might be logged as separate issue.

**Review by damyanpetev — APPROVED** [review]

**Review by zdrawku — APPROVED** [review]

**Pull request merged**